packs/outline/README.md

```diff
@@ -0,0 +1,273 @@
+# Outline
+
+This pack contains a service job that runs [Outline](https://www.getoutline.com/) in a single Nomad client. It currently supports
+being run by the [Docker](https://www.nomadproject.io/docs/drivers/docker) driver.
+
+It has 4 tasks:
+- **Outline:** [*(reference)*](https://www.getoutline.com/) the wiki and knowledgebase for teams;
+- **PostgreSQL:** [*(reference)*](https://www.postgresql.org/) the persistent database used by Outline;
+- **Redis:** [*(reference)*](https://redis.io/) for in-memory storage and cache;
+- **MinIO:** [*(reference)*](https://min.io/) for images storage, compatible with S3 cloud storage service.
+
+Setup:
+- Service-to-service communication is handled with Consul Connect;
+- Bitnami images are being used for [PostgreSQL](https://hub.docker.com/r/bitnami/postgresql), [Redis](https://hub.docker.com/r/bitnami/redis) and [MinIO](https://hub.docker.com/r/bitnami/minio). They are non-root containers and as such the mounted files and directories must have the proper permissions for the UID 1001. To achieve this, these 3 images are being run as user 1001 and there are prestart tasks that create their respective directories, while setting the correct ownership, in the host filesystem with the raw exec driver;
+- Only MinIO and Outline ports are exposed. To simplify the pack, MinIO has a static host port of 9000, whereas Outline's port is being randomly assigned. In production it's recommended that none of the services are directly exposed and that traffic is run through a reverse proxy that supports Consul Connect enabled services, like Traefik;
+- MinIO's group has a poststart task that applies a policy that disables directory browsing. It does it by finding the correct Docker container, exec'ing into the container and applying the policy with MinIO's MC;
+- Outline supports a number of authentication methods. Slack is used for demonstration purposes, but enabling more authentication methods is a matter of defining the [environment variables](https://github.com/outline/outline/blob/main/.env.sample);
+- The default variables are defined for a local setup.
+
+## Requirements
+Clients that expect to run this job require:
+- [Docker volumes](https://www.nomadproject.io/docs/drivers/docker "Docker volumes") to be enabled within their Docker plugin stanza:
+```hcl
+plugin "docker" {
```

Comment:
Thanks for documenting this requirement.  I ran into an issue though, where I had some nodes that had volumes enabled and some that didn't.  Would you be willing to add `constraints`[https://www.nomadproject.io/docs/job-specification/constraint] to the tasks that have to be run on a client with docker volumes enabled?  I think this type of mixed client environment is pretty common, so it would be really helpful.

Reply:
Sure! I'll submit the improvement by EOD.
I'll add a constraint for both the raw_exec driver and Docker volumes.

Thanks for the feedback!

Reply:
Hello! As suggested, the new commit adds a new constraints variable that defaults to:

`  constraint {
    attribute = "${attr.driver.docker.volumes.enabled}"
    value     = "true"
  }
`